sdk/cosmos/azure-cosmos/azure/cosmos/aio/_container.py

```diff
@@ -365,27 +437,27 @@ def query_items(
                 :caption: Parameterized query to get all products that have been discontinued:
                 :name: query_items_param
         """
-        feed_options = _build_options(kwargs)
         response_hook = kwargs.pop('response_hook', None)
-        max_item_count = kwargs.pop('max_item_count', None)
+        if session_token is not None:
+            kwargs['session_token'] = session_token
+        if initial_headers is not None:
+            kwargs['initial_headers'] = initial_headers
+        if priority_level is not None:
+            kwargs['priority_level'] = priority_level
+        feed_options = _build_options(kwargs)
         if max_item_count is not None:
             feed_options["maxItemCount"] = max_item_count
-        populate_query_metrics = kwargs.pop('populate_query_metrics', None)
         if populate_query_metrics is not None:
             feed_options["populateQueryMetrics"] = populate_query_metrics
-        populate_index_metrics = kwargs.pop("populate_index_metrics", None)
         if populate_index_metrics is not None:
             feed_options["populateIndexMetrics"] = populate_index_metrics
-        enable_scan_in_query = kwargs.pop('enable_scan_in_query', None)
         if enable_scan_in_query is not None:
             feed_options["enableScanInQuery"] = enable_scan_in_query
-        partition_key = kwargs.pop('partition_key', None)
         if partition_key is not None:
             feed_options["partitionKey"] = self._set_partition_key(partition_key)
```

Comment:
Do we need await here?

Reply:
This is a sync (pageable) method, so we can't await here.
This is awaited further down the query pipeline when we fetch the first page:
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/cosmos/azure-cosmos/azure/cosmos/aio/_query_iterable_async.py#L101-L102